client/src/app/auth/services/auth.service.ts

```diff
@@ -0,0 +1,39 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { tap } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  constructor(private http: HttpClient) {}
+
+  login(email: string, password: string) {
+    return this.http
+      .post('http://localhost:3000/users/login', {
+        email,
+        password,
+      })
+      .pipe(tap((res) => this.setSesstion(res)));
+  }
+
+
+  isLogged() {
+    const expiresIn = localStorage.getItem('expiresIn');
+    if (expiresIn) {
+      return Date.now() < Number(expiresIn);
+    }
+    return false;
+  }
+
+  logout() {
+    localStorage.removeItem('expiresIn');
+    localStorage.removeItem('idToken');
+  }
+
+  private setSesstion(res: any) {
```

Comment:
Please do no use `any`

Reply:
Should I provide all the possibles types it receives if I do not use any in this case. I tried to provide all the types but it did not compile, when I log type of res, it showed as an object but if I put object as "res: object", it is not compiling.